lib/recent.go

```diff
@@ -0,0 +1,107 @@
+package lib
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type RecentFiles struct {
+	Terraform []string `json:"terraform"`
+	OpenTofu  []string `json:"openTofu"`
+}
+
+func addRecent(requestedVersion string, installPath string, distribution string) {
+	if !validVersionFormat(requestedVersion) {
+		logger.Errorf("The version %q is not a valid version string and won't be stored", requestedVersion)
+		return
+	}
+	installLocation := GetInstallLocation(installPath)
+	recentFilePath := filepath.Join(installLocation, recentFile)
+	var recentFileData RecentFiles
+	if CheckFileExist(recentFilePath) {
+		unmarshalRecentFileData(recentFilePath, &recentFileData)
+	}
+	prependRecentVersionToList(requestedVersion, distribution, &recentFileData)
+	saveRecentFile(recentFileData, recentFilePath)
+}
+
+func prependRecentVersionToList(version, distribution string, r *RecentFiles) {
+	var sliceToCheck []string
+	if distribution == distributionTerraform {
+		sliceToCheck = r.Terraform
+	} else if distribution == distributionOpenTofu {
+		sliceToCheck = r.OpenTofu
+	}
```

Comment:
Wouldn't `switch` be a bit more streamlined here? 🤔

Reply:
Changed to switch